### PR TITLE
Use bind mounts in sample config

### PR DESCRIPTION
**Pull request by casperklein**
# Description

If we switch to bind mounts in the sample config, that will simplify backup/restore of DMS.

See also: #1906

This would also make the not 100% correct working script at https://docker-mailserver.github.io/docker-mailserver/edge/faq/#what-about-backups obsolete.

Backup/restore would then just be simple host commands `cp / mv`..

What do you think?

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (README.md or the documentation under `docs/`)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by georglauterbach** [comment]
Actually a good idea. But this requires a documentation update.

**Comment by casperklein** [comment]
For sure. But before wasting time, I wanted to know beforehand what your opinions are 😎

**Comment by wernerfred** [comment]
I prefer bind mounts over volumes (personal opinion) so you have my vote

File: docs/content/faq.md [hunk 1]
```diff
@@ -54,16 +54,22 @@ Please do not use `CRLF`.
 
 ### What about backups?
 
+#### Bind mounts (default)
+
+```bash
+tar czf backup-$(date +%F).tar.gz config data
+```
+
```
Comment (polarathene) [inline docs/content/faq.md:62]:
````suggestion
#### Bind mounts (default)

From the location of your `docker-compose.yml`, create a compressed archive of your `./config` and `./data` folders:

```bash
tar --gzip -cf "backup-$(date +%F).tar.gz" config data
```

Then to restore `./config` and `./data` folders from your backup file:

```bash
tar -xf backup-date.tar.gz
```

````

**Review by polarathene — CHANGES_REQUESTED** [review]
Might be worth adding to the FAQ how to restore?

I also wanted to be a bit more explicit about the option for compressing to gzip. I'd suggest `--zstd` with `tar.zst` but not sure how wide the support for that is.

**Comment by polarathene** [comment]
> Backup/restore would then just be simple host commands `cp / mv`..

Doesn't that mess with permissions / ownership metadata on files?

I prefer bind mounts myself, but a common issue I can run into with containers is the container writes the files with different uid/gid (mostly only an annoyance if you're non-root), or you configure some files from the host side and if the container lacks permission to read/write the file it gets upset.

Regarding the referenced issue, kinda seems odd to backup clamav bloat, depending on the age of the backup when it's restored, I'm not sure how worthwhile that would be to retain? This is still a nice improvement anyway :)

**Comment by georglauterbach** [comment]
> Might be worth adding to the FAQ how to restore?
> 
> I also wanted to be a bit more explicit about the option for compressing to gzip. I'd suggest `--zstd` with `tar.zst` but not sure how wide the support for that is.

I think creating a `.tar.gz` is the most widely compatible archive for Linux and we should use that:)

**Comment by polarathene** [comment]
> I think creating a `.tar.gz` is the most widely compatible archive for Linux and we should use that :)

Yeah, that's why I was reluctant with my preference to use zstd (_which is definitely better when available_).

I'd still prefer the more explicit option of `--gzip` however, since this is a FAQ item and the user is probably going to copy/paste anyway.

**Comment by wernerfred** [comment]
> I'd still prefer the more explicit option of `--gzip` however

I prefer verbosity, too. Always hard for future contributions to look up / follow along if a bunch of short options is thrown together using a tool a contributor might not know. As most will copy/paste it anyways it should not be a big deal

**Comment by georglauterbach** [comment]
Yes. We will use the long option `--gzip` :)

**Comment by casperklein** [comment]
@docker-mailserver/maintainers feel free to edit/add docu yourself to this PR 👍

**Review by georglauterbach — APPROVED** [review]

File: docs/content/faq.md [hunk 2]
```diff
@@ -65,7 +65,7 @@ tar --gzip -cf "backup-$(date +%F).tar.gz" config data
 Then to restore `./config` and `./data` folders from your backup file:
 
 ```bash
-tar -xf backup-date.tar.gz
+tar --gzip -xf backup-date.tar.gz
```
Comment (polarathene) [inline docs/content/faq.md:68]: [`--gzip` hint isn't needed](https://askubuntu.com/a/398336) for extraction.

Comment (casperklein) [reply docs/content/faq.md:68]: You are right. But it doesn't hurt if you ask me. I prefer calling tools with explicit parameters, especially in documentation. Feel free to remove it 👍

**Review by polarathene — APPROVED** [review]

**Comment by NorseGaud** [comment]
<3

**Pull request merged**